tests/test_connector.rs

```diff
@@ -586,6 +591,262 @@ fn test_ft_transfer_call_fee_greater_than_amount() {
     assert_eq!(balance, 0);
 }
 
+fn call_deposit_with_proof(
+    account: &UserAccount,
+    contract: &str,
+    proof: &str,
+) -> Vec<Option<ExecutionResult>> {
+    let proof: Proof = serde_json::from_str(proof).unwrap();
+    let res = account.call(
+        contract.to_string(),
+        "deposit",
+        &proof.try_to_vec().unwrap(),
+        DEFAULT_GAS,
+        0,
+    );
+    res.promise_results()
+}
+
+fn call_set_paused_flags(
+    account: &UserAccount,
+    contract: &str,
+    paused_mask: PausedMask,
+) -> ExecutionResult {
+    let res = account.call(
+        contract.to_string(),
+        "set_paused_flags",
+        &paused_mask.try_to_vec().unwrap(),
+        DEFAULT_GAS,
+        0,
+    );
+    res
+}
+
+fn create_user_account(master_account: &UserAccount) -> UserAccount {
+    let user_account = master_account.create_user(
+        "eth_recipient.root".to_string(),
+        to_yocto("100"), // initial balance
+    );
+    user_account
+}
+
+#[test]
+fn test_admin_controlled_only_admin_can_pause() {
+    let (master_account, contract) = init(CUSTODIAN_ADDRESS);
+    let user_account = create_user_account(&master_account);
+
+    // Try to pause from the user - should fail
+    let res = call_set_paused_flags(&user_account, CONTRACT_ACC, PAUSE_DEPOSIT);
+    let promises = res.promise_results();
+    let p = promises[1].clone();
+    match p.unwrap().status() {
+        ExecutionStatus::Failure(_) => {}
```

Comment:
Discussed privately that matching on the error message is currently not possible because of some linker error.

It is ok to merge as is, but please file and issue and add a TODO comment with the issue number so we remember to come back to it in the future.

Reply:
As @mrLSD introduced another way to check for the error message in #106 ([code](https://github.com/aurora-is-near/aurora-engine/blob/dc245406a6973b22baddd6ba2cae949a168c7b88/tests/test_connector.rs#L868)), the same way will be used when that PR is merged.

Reply:
Added explicit error message check.